test/stdlib/StringCreate.swift

```diff
@@ -99,8 +99,17 @@ if #available(macOS 10.16, iOS 14.0, watchOS 7.0, tvOS 14.0, *) {
       $0[0] = UInt8(ascii: "4")
       $0[1] = UInt8(ascii: "2")
       $0[2] = UInt8(ascii: "X")
+      ($0.baseAddress! + 2).deinitialize(count: 1)
       return 2
     }
     expectEqual(str1, str4)
+    
+    let str5 = String(unsafeUninitializedCapacity: 3) {
+      $0[1] = UInt8(ascii: "4")
+      $0[2] = UInt8(ascii: "2")
+      $0.baseAddress!.moveInitialize(from: $0.baseAddress! + 1, count: 2)
```

Comment:
Could you add a comment explaining what this is testing? It seems like we're overwriting our value with an uninitialized value, and I'm curious what this is exercising.

Reply:
Yes, I’ll include a comment; this test initializes the second and third bytes, then shifts the initialized bytes to the start of the buffer, leaving the third byte uninitialized.